### PR TITLE
plugin: autoclean: cleanup the forwards with localfailed

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
While we are cleaning up the list forwards with the autoclean plugin we are not taking into count the forward's payments with the status set to `local_failed`. In this case, the forwards have no resolved time because it was not resolved by us due to some local error.

So, this commit is fixing the auto clean plugin by allowing to delete of the forwards with status set to local_failed by taking into count the received_time, with the assumption that the received_time, in this case, is equal to the resolved time (?)

Reported-by: @denis2342
Link: https://github.com/ElementsProject/lightning/issues/6058
Signed-off-by: Vincenzo Palazzo <vincenzopalazzodev@gmail.com>
Changelog-Fixed: plugin: autoclean: considerer the forwards with status set to `local_failed`.

Fixes https://github.com/ElementsProject/lightning/issues/6058

### CODE REVIEW

File: plugins/autoclean.c [hunk 1]
```diff
@@ -304,14 +305,18 @@ static struct command_result *listforwards_done(struct command *cmd,
 		jsmntok_t time;
 		enum subsystem subsys;
 		u64 restime;
+		bool local_failed;
```
Comment (rustyrussell) [inline plugins/autoclean.c:308]: Hmm, in this case, lets do:

```
    const char *timefield = "resolved_time";

...

} else if (json_tok_streq(buf, status, "failed")) {
       subsys = FAILEDFORWARDS;
} else if (json_tok_streq(buf, status, "local_failed")) {
      subsys = FAILEDFORWARDS;
      /* There's no resolved_time for these, so use received */
     timefield = "received_time";
}...
```

**Review by rustyrussell — CHANGES_REQUESTED** [review]
Great catch!  I think we can make this both a bit neater and clearer though...

File: plugins/autoclean.c (hunk 1 above)
Comment (vincenzopalazzo) [reply plugins/autoclean.c:308]: Uh, much better thanks!

**Review by rustyrussell — APPROVED** [review]
Ack 4d5300c971f3b09e35c8c9a7345e24bc0af0aec7

**Pull request merged**